src/overlays/actors/ovl_En_Wonder_Talk2/z_en_wonder_talk2.c

```diff
@@ -26,19 +31,275 @@ const ActorInit En_Wonder_Talk2_InitVars = {
+                // Processing range
+                osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 処理範囲               %f\n" VT_RST, this->unk_15C);
+                switch (this->unk_154) {
+                    case 0:
+                        // Normal
+                        osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆ 通常 ☆☆ \n" VT_RST);
+                        break;
+                    case 2:
+                        // Check only
+                        osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆ チェックのみ ☆☆ \n" VT_RST);
+                        break;
+                    case 3:
+                        // Lock only
+                        osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆ ロックのみ ☆☆ \n" VT_RST);
+                        break;
+                }
+            }
+
+            this->unk_158 = 0;
+            func_8002F1C4(&this->actor, globalCtx, this->unk_15C + 50.0f, 100.0f, 0);
+        }
+    }
+}
+
+void func_80B3A3D4(EnWonderTalk2* this, GlobalContext* globalCtx) {
+    u8 dialogState;
+
+    if (BREG(2)) {
+        // Oh
+        osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ わー %d\n" VT_RST, func_8010BDBC(&globalCtx->msgCtx));
+    }
+    dialogState = func_8010BDBC(&globalCtx->msgCtx);
+
+    switch (dialogState) {
+        case 5:
+        case 6:
+            if (func_80106BC8(globalCtx)) {
+                if (func_8010BDBC(&globalCtx->msgCtx) == 5) {
+                    func_80106CCC(globalCtx);
+                }
+            } else {
+                break;
+            }
+        case 0:
+            if ((this->switchFlag >= 0) && (this->unk_154 != 4)) {
+                Flags_SetSwitch(globalCtx, this->switchFlag);
+                // (Forced) I saved! It's all about it!
+                osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ (強制)セーブしたよ！おもいっきり！ %x\n" VT_RST, this->switchFlag);
+            }
+
+            if (this->unk_154 == 4) {
+                this->unk_15A = 1;
+            }
+            this->actor.flags &= -0x12;
+            func_8002DF54(globalCtx, NULL, 7U);
+            this->unk_156 = 1;
+            this->actionFunc = func_80B3A4F8;
+            break;
+    }
+}
+
+void func_80B3A4F8(EnWonderTalk2* this, GlobalContext* globalCtx) {
+    Player* player;
+
+    player = PLAYER;
+    this->unk_158++;
+    if ((this->switchFlag >= 0) && (Flags_GetSwitch(globalCtx, this->switchFlag) != 0)) {
+        if (this->unk_15A == 0) {
+            this->actor.flags &= ~1;
+            this->unk_15A = 1;
+            return;
+        }
+    } else {
+        if ((this->unk_154 != 4) || (this->unk_15A == 0)) {
+            if (BREG(2) != 0) {
+                // distance
+                osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ きょり %f\n" VT_RST, this->actor.xzDistFromLink);
+            }
+            if (((this->actor.xzDistFromLink < (40.0f + this->unk_15C)) &&
+                 (fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) < 100.0f)) &&
+                (Gameplay_InCsMode(globalCtx) == 0)) {
```

Comment:
This function returns 0 or 1 so you can do !GamePlay_InCsMode... instead of == 0 I think